printer.go

```diff
@@ -9,22 +9,35 @@ import (
 	"io"
 	"math"
 	"os"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/gonum/floats"
 )
 
+var printerHeadings = [...]string{
+	"Iter",
+	"Runtime",
+	"FuncEvals",
+	"Func",
+	"GradEvals",
+	"|Gradient|∞",
+	"HessEvals",
+}
+
+const (
+	printerBaseTmpl = "%6v  %12v  %9v  %22v" // Base template for headings and values that are always printed.
```

Comment:
The overall intention was to have a fixed-width output for the majority of runs.

Width of 6 for the iteration may be slightly excessive, I can make it 5 or even 4 if you want.

Width of 12 is just an empirical observation of the maximum string length that the default format for Time gives.

Width of 9 for the number of evaluations is simply the width of the headings `FuncEvals`, ...  It is excessive, so if you can think of a shorter, descriptive heading, I will replace it.

Width of 22 for floats is again an empirical observation: 16 digits + sign, dot, e, sign of the exponent and 2 digits of the exponent. We could use a real fixed-width format like %12.5e but %v has proved useful for debugging in the past.